cicle_intersects_triangle

prb616 opened the pull request:
# Description

This pull request adds a new usage example for the circle_triangle_intersect function in SplashKit.
The example demonstrates how a circle (controlled by the user via arrow keys) can intersect with a fixed triangle on the screen.

## Type of change

## How Has This Been Tested?
The program was tested by:
- Running all 4 language versions (C++, Python, C# top-level, C# OOP)
- Moving the circle toward the triangle using arrow keys
- Confirming color and text update as expected upon intersection
- Confirming no errors on startup or during gameplay

## Testing Checklist

- [ ] Tested in latest Chrome
- [ ] Tested in latest Firefox
- [ ] npm run build
- [ ] npm run preview

## Checklist

_Please delete options that are not relevant._

### If involving code

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

## Folders and Files Added/Modified

- Added:
  - [ ] usage-examples/geometry/circle_triangle_intersect-1-example-top-level.cs
  - [ ] usage-examples/geometry/circle_triangle_intersect-1-example.cpp
  - [ ] usage-examples/geometry/circle_triangle_intersect-1-example-oop.cs
  - [ ] usage-examples/circle_triangle_intersect-1-example.py
  - [ ] usage-examples/geometry/circle_triangle_intersect-1-example.png

CATHoDE33 reviewed (changes requested):
Hey Prabhjot, after going through everything I've come up with the following. Please make these changes in addition to the code changes and I'll have a second look:

- [ ] C# OOP and Python programs don't work at all due to multiple code issues, please fix this and resubmit
- [ ] File naming scheme needs to be 'circle_triangle_intersect-1-example' instead of 'cicle_intersects_triangle-1-example'. For the C++ file, it needs to end in 'example-sk.cpp'
- [x] Show some simple on-screen instructions for the user, such as 'Use the arrow keys'
- [x] Since this is an interactive program, please submit a GIF file to show the program working instead of a PNG image
- [ ] .txt file hasn't been submitted. This is simply a text file that contains a basic description of the program (fixed triangle and interactive circle for example)

Please note, I have only requested changes in the OOP version, but please apply the relevant changes to every version of the program

CATHoDE33 commented on public/usage-examples/geometry/cicle_intersects_triangle-1-example-opp.cs:
```diff
@@ -0,0 +1,46 @@
+using SplashKitSDK;
+
```
Program can't run due to missing namespace. Insert:


```suggestion
namespace CircleTriangleIntersectExample
{
```

CATHoDE33 commented on public/usage-examples/geometry/cicle_intersects_triangle-1-example-opp.cs:
```diff
@@ -0,0 +1,46 @@
+    public static void Main()
+    {
+        // A fixed triangle is placed as the target.
+        Window gameWindow = new Window("Circle-Triangle Intersect", 600, 600);
```
For programs that only require a single window, you can create one without creating a variable for it:

```suggestion
        SplashKit.OpenWindow("Circle Triangle Intersect", 600, 600);
```

CATHoDE33 commented on public/usage-examples/geometry/cicle_intersects_triangle-1-example-opp.cs:
```diff
@@ -0,0 +1,46 @@
+            Center = SplashKit.PointAt(100, 100)
+        };
+
+        while (!gameWindow.CloseRequested)
```
```suggestion
        while (!SplashKit.QuitRequested())
```

CATHoDE33 commented on public/usage-examples/geometry/cicle_intersects_triangle-1-example-opp.cs:
```diff
@@ -0,0 +1,46 @@
+                SplashKit.DrawText("Circle is Intersecting the triangle", Color.Black, 170, 20);
+            }
+
+            SplashKit.RefreshScreen(60);
```
```suggestion
            SplashKit.RefreshScreen();
         }
         SplashKit.CloseAllWindows();
```

CATHoDE33 commented on public/usage-examples/geometry/cicle_intersects_triangle-1-example-opp.cs:
```diff
@@ -0,0 +1,46 @@
+        {
+            SplashKit.ProcessEvents();
+
+            if (SplashKit.KeyDown(KeyCode.UpKey)) playerCircle.Center.Y -= 2;
```
The Thoth Tech style guide says that all 'if' statements need to be on their own line. Do this for all four key input lines:
```suggestion
            if (SplashKit.KeyDown(KeyCode.UpKey))
                {
                    circleShape.Center.Y -= 2;
                }
```

prb616 commented:
> Hey Prabhjot, after going through everything I've come up with the following. Please make these changes in addition to the code changes and I'll have a second look:
> 
> * [ ]  C# OOP and Python programs don't work at all due to multiple code issues, please fix this and resubmit
> * [ ]  File naming scheme needs to be 'circle_triangle_intersect-1-example' instead of 'cicle_intersects_triangle-1-example'. For the C++ file, it needs to end in 'example-sk.cpp'
> * [ ]  Show some simple on-screen instructions for the user, such as 'Use the arrow keys'
> * [ ]  Since this is an interactive program, please submit a GIF file to show the program working instead of a PNG image
> * [ ]  .txt file hasn't been submitted. This is simply a text file that contains a basic description of the program (fixed triangle and interactive circle for example)
> 
> Please note, I have only requested changes in the OOP version, but please apply the relevant changes to every version of the program


i corrected all the codes and changed the name as per naming scheme, added gif of output and added txt file

CATHoDE33 commented:
Please complete the above code changes as well as:
-Some of the committed files are still named 'cicle_intersects_triangle-1-example'. Make sure circle is spelled correctly and commit again
-Delete the original .png file from your commit since you've now made a .gif file

I'll have another look once everything is done

prb616 commented:
@CATHoDE33, thanks for your review, I have done all the changes that you asked for. Please have a look and let me know if you need any other changes.

CATHoDE33 commented:
I've had a second look and there are many changes that need to be made, too many to mention at once. First I will note the following and when they're completed I'll have another look and make some more requests.

- [ ] C++ version:
No on-screen instructions
Circle moves too slowly

- [ ] Python version:
Shows errors in the terminal when running
No on-screen instructions
Circle moves way too fast

- [ ] C# OOP version:
This version doesn't run at all

- [ ] C# top level version:
Circle moves way too fast

- [ ] gif. file needs to be cropped better. You can see past the sides of the window

- [ ] txt. file hasn't been changed at all since I last requested you to do so. Please do this

- [ ] All files are still not named correctly, they need to be variations of 'circle_triangle_intersect-1-example'
